src/Sylius/Bundle/ApiBundle/Validator/Constraints/ShopUserResetPasswordTokenExistsValidator.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
```

Comment:
What do you think about moving it to the user bundle?

Reply:
Yes you are totally right, i thought about it during the dev and I don't understand why they don't extends some of the existing models created to handle password reset request for example.
@lchrusciel is their a reason for that or it's just an oversight ?

Reply:
Those classes are only usable inside the ApiBundle because the `input` command is taking both GET and POST data, on the contrary CoreBundle have validators used to only validate forms.

So finally I don't think it's very useful to move those new validators to the CoreBundle (except for `ShopUserExists` and `ShopUserExistsValidator`).

Reply:
@rimas-kudelis if you follow what the UI is doing with the token you will see it will never be part of a form and never be validated. I could introduce a new way of validating this data but it's not the purpose of this PR.

Reply:
But it's not about what the UI is or isn't doing. Your validators are fully usable outside the API bundle, and that's all that matters IMO. They're universal, so why put them in the API bundle? The UI may or may not use these validators, but the point of moving them is not that they **are** used by the UI, but that they **can** be used.

I'm also kinda surprised that you suggested moving them to CoreBundle instead of UserBundle and @vvasiloi  agreed with that. As I understand it, CoreBundle holds integrations between other Sylius bundles, but both the repositories and the fields involved in these validators are specified in the classes from UserBundle, so I so far fail to understand why CoreBundle (or ApiBundle) and not UserBundle would be the right place for them.

Reply:
Oh wait, I get it now. Core is the part where User splits into ShopUser and AdminUser. I take all my previous comment back.

Reply:
Answers are crossing 😉